src/models/stores/serial.ts

```diff
@@ -0,0 +1,98 @@
+import { NodeChannelInfo } from "src/plugins/dataflow-tool/model/utilities/node";
+
+export class SerialDevice {
+    value: string;
+    localBuffer: string;
+    private port: any; // SERIAL TODO: types SerialPort | null 
+
+    constructor() {
+      this.value = '0';
+      this.localBuffer = '';
+    }
+
+    public hasPort(){
+      return this.port !== undefined;
+    }
+  
+    public async requestAndSetPort(){
+      try {
+          await (navigator as any).serial.requestPort()
+          .then((p: any) => {
+            this.port = p;
+          });
```

Comment:
As I mentioned above, the types just worked for me so I was able to eliminate the `as any`. Secondarily, using `await` and `then` in the same statement is odd. I would think something like this should work:
```suggestion
          this.port = await navigator.serial.requestPort();
```